### PR TITLE
Created a responsive footer

### PR DESCRIPTION
**Pull request by Weeezzt**
This PR introduces a new, responsive `Footer` component for the application. It includes navigation links, social media icons, and contact information to enhance the website's usability and visual appeal.

#### **Key Features:**
- **Navigation Section**: 
  - Links for Home, About, and Contact. *Placeholders 
- **Social Media Section**: 
  - Interactive icons for GitHub, Instagram, and TikTok using `react-icons`.
- **Contact Information Section**: 
  - Address, phone number, and email for easy communication. *Placeholder information

#### **Styling Highlights:**
- Responsive design with `flex` utilities for stacking and spacing.
- Adaptive layout:
  - Vertical on smaller screens.
  - Horizontal on larger screens with flexible gaps (`sm:gap-8`, `md:gap-16`, etc.).
- Dark background (`bg-gray-800`) with white text for contrast and readability.

#### **Testing and Verification:**
- Ensured responsiveness across multiple screen sizes.
- Verified accessibility for screen readers.
- Checked alignment and spacing consistency.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Footer` component, enhancing the layout with navigation, social media links, and contact information.
	- Added a new `FootermenuItems` component for rendering navigation links in the footer.
	- Updated the `Header` component to improve navigation functionality with responsive design.

- **Bug Fixes**
	- Adjusted the layout structure of the `App` component for better responsiveness.

- **Documentation**
	- Enhanced accessibility and usability of the footer and header components.

### CODE REVIEW

File: frontend/src/components/common/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+              <li>About</li>
+            </a>
+            <a href="/">
+              <li>Contact</li>
+            </a>
+          </ul>
+        </div>
+        <div className="">
+          <h2 className="text-xl font-bold mb-2">Social Media</h2>
+          <ul className="text-3xl flex gap-4">
```
Comment (MickeMarten) [inline frontend/src/components/common/Footer.tsx:25]: Consider using justify-items-evenly if we dont not want the items in the center and more evenly around the footer.

**Review by MickeMarten — DISMISSED** [review]
This pr is merged with header component, should we remove the Header PR and rename this PR?

Both header and footer components are responsive and reusable.

In further development in the future we might make dropdown items and links more reusable by using a object we can loop through and create elements inside the loop. I think its mroe maintainble that way as our application grows.

Good job!

**Review by MikaelaMathisson — CHANGES_REQUESTED** [review]
I think the code looks clean and very easily readable. 

I saw that coderabbit suggested a change in package.json where there are dublicates.
 Look into that feedback before merge? And also look into the feedback from other review about maybe renaming this PR.
 Other than that i think this looks great.

File: frontend/src/components/common/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { FaGithub } from "react-icons/fa";
+import { FaTiktok } from "react-icons/fa";
+
+export default function Footer() {
+  return (
+    <div className="flex sm:flex-row flex-col w-full min-h-52 bg-gray-800 text-white pl-10 space-y-6 sm:space-y-0 sm:justify-center sm:gap-8 md:gap-16 lg:gap-24 xl:gap-32 2xl:gap-40 3xl:gap-48 py-10">
+      <div className="min-[520px]:flex min-[520px]:justify-between min-[520px]:pr-16 sm:pr-0 sm:gap-8 md:gap-16 lg:gap-24 xl:gap-32 2xl:gap-40 3xl:gap-48">
+        <div className="mb-5 sm:mb-0">
+          <h2 className="text-xl font-bold mb-2">Navigering</h2>
+          <ul className="">
```
Comment (MatsJoh50) [inline frontend/src/components/common/Footer.tsx:11]:
```
const menuItems: { title: string, url: string }[] = [
  { title: "Home", url: "#" },
  { title: "About", url: "#" },
  { title: "Contact", url: "#" },
]


const RenderMenuItems = () => {
  return (
        <ul className="">
          {menuItems.map((menu, index) => (
              <li
                  key={index}
                  className=""
              >
                <a href={menu.url} className="">
                  {menu.title}
                </a>
              </li>
          ))}
        </ul>
  );
};
```

**Review by MatsJoh50 — CHANGES_REQUESTED** [review]
As with header, great start for a blank canvas for headers and footers.
and same example here,

please convert the Nav list to  .map() for ease of update.

we might also want to push the list back to layout and send it down as a props, while this is spreading out the code we are making it more consice to lower risk of mistakes.

File: frontend/src/components/common/Footer.tsx (hunk 2 above)
Comment (AlexanderK88) [reply frontend/src/components/common/Footer.tsx]: We have implemented the requested change

**Comment by AlexanderK88** [comment]
> I think the code looks clean and very easily readable.
> 
> I saw that coderabbit suggested a change in package.json where there are dublicates. Look into that feedback before merge? And also look into the feedback from other review about maybe renaming this PR. Other than that i think this looks great.

We have implemented the requested change

**Review by MatsJoh50 — DISMISSED** [review]
Good changes after suggestions.

**Review by MikaelaMathisson — APPROVED** [review]

**Review by MatsJoh50 — APPROVED** [review]
We are conitnuing the path of a great shell to start a product with.

**Pull request merged**